From f2815ea4de844c4287b7a61da26c2b79e7835faa Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 24 Mar 2025 11:49:12 +0100
Subject: [PATCH 879/954] feat(st-pmic): display information only in BL2

Display the STPMIC version and product ID only in BL2.

No need to be displayed in BL31, when STPMIC drvier is used to switch off
the DDR power supplies.

This patch also move the 'pmic2->ref_id' initialization in
initialize_pmic_i2c().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git a6368e4a15afea236e2bc92e880a35997f3a92b9 feat(st-pmic): display information only in BL2]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ic414c735d16e2937d1435918843d3fad20e8bde5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/446189
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 drivers/st/pmic/stm32mp_pmic2.c | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/drivers/st/pmic/stm32mp_pmic2.c b/drivers/st/pmic/stm32mp_pmic2.c
index c814757b2d..a496b78850 100644
--- a/drivers/st/pmic/stm32mp_pmic2.c
+++ b/drivers/st/pmic/stm32mp_pmic2.c
@@ -131,6 +131,7 @@ static int dt_pmic2_i2c_config(struct dt_node_info *i2c_info,
 bool initialize_pmic_i2c(void)
 {
 	int ret;
+	uint8_t val;
 	struct dt_node_info i2c_info;
 	struct i2c_handle_s *i2c = &i2c_handle;
 	uint32_t i2c_addr = 0U;
@@ -178,6 +179,12 @@ bool initialize_pmic_i2c(void)
 	pmic2->i2c_handle = &i2c_handle;
 	pmic2->i2c_addr = i2c_addr;
 
+	if (stpmic2_get_product_id(pmic2, &val) != 0) {
+		ERROR("Failed to access PMIC\n");
+		panic();
+	}
+	pmic2->ref_id = ((val & PMIC_REF_ID_MASK) >> PMIC_REF_ID_SHIFT);
+
 	return true;
 }
 
@@ -544,19 +551,14 @@ void initialize_pmic(void)
 		return;
 	}
 
+#if IMAGE_BL2
 	if (stpmic2_get_version(pmic2, &val) != 0) {
 		ERROR("Failed to access PMIC\n");
 		panic();
 	}
 	INFO("PMIC2 version = 0x%02x\n", val);
-
-	if (stpmic2_get_product_id(pmic2, &val) != 0) {
-		ERROR("Failed to access PMIC\n");
-		panic();
-	}
-	INFO("PMIC2 product ID = 0x%02x\n", val);
-
-	pmic2->ref_id = ((val & PMIC_REF_ID_MASK) >> PMIC_REF_ID_SHIFT);
+	INFO("PMIC2 product ID = 0x%02x\n", pmic2->ref_id);
+#endif
 
 	ret = register_pmic2();
 	if (ret < 0) {
-- 
2.53.0

